Thanks @Foxboron, I skipped those tests as I couldn't find the root cause. Unfortunately, I got stuck on another set of failing tests that aren't clear to me either.

```
ok      github.com/foxboron/ssh-tpm-agent/contrib       0.002s
--- FAIL: TestNokey (0.00s)
    keyring_test.go:37: err: no such file or directory
--- FAIL: TestRemoveKey (0.00s)
    keyring_test.go:64: we can still read the key
--- FAIL: TestNokeyThreaded (0.00s)
    threadkeyring_test.go:45: err: no such file or directory
--- FAIL: TestRemoveKeyThreaded (0.00s)
    threadkeyring_test.go:75: we can still read the key
FAIL
FAIL    github.com/foxboron/ssh-tpm-agent/internal/keyring      0.004s
```

--- a/pkgs/by-name/ss/ssh-tpm-agent/package.nix
+++ b/pkgs/by-name/ss/ssh-tpm-agent/package.nix
@@ -3,27 +3,39 @@
   buildGoModule,
   fetchFromGitHub,
   nix-update-script,
+  openssh,
   openssl,
 }:
 
 buildGoModule rec {
   pname = "ssh-tpm-agent";
-  version = "0.7.0";
+  version = "0.8.0";
 
   src = fetchFromGitHub {
     owner = "Foxboron";
     repo = "ssh-tpm-agent";
-    rev = "v${version}";
-    hash = "sha256-yK7G+wZIn+kJazKOFOs8EYlRWZkCQuT0qZfmdqbcOnM=";
+    tag = "v${version}";
+    hash = "sha256-CSxZctiQ/d4gzCUtfx9Oetb8s0XpHf3MPH/H0XaaVgg=";
   };
 
   proxyVendor = true;
 
-  vendorHash = "sha256-njKyBfTG/QCPBBsj3Aom42cv2XqLv4YeS4DhwNQNaLA=";
+  vendorHash = "sha256-84ZB1B+RczJS08UToCWvvVfWrD62IQxy0XoBwn+wBkc=";
 
   buildInputs = [
     openssl
   ];
+
+  nativeCheckInputs = [
+    openssh
+  ];
+
+  # disable broken tests, see https://github.com/NixOS/nixpkgs/pull/394097
+  preCheck = ''
+    rm cmd/scripts_test.go
+    substituteInPlace internal/keyring/keyring_test.go --replace-fail ENOKEY ENOENT
+    substituteInPlace internal/keyring/threadkeyring_test.go --replace-fail ENOKEY ENOENT
+  '';
 
   passthru.updateScript = nix-update-script { };
 

